src/modules/fancyzones/lib/WindowMoveHandler.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+interface IFancyZonesSettings;
+interface IZoneWindow;
+
+class WindowMoveHandler
+{
+public:
+    WindowMoveHandler(const winrt::com_ptr<IFancyZonesSettings>& settings);
+    ~WindowMoveHandler();
+
+    bool InMoveSize() const noexcept;
+    bool IsDragEnabled() const noexcept;
+
+    void MoveSizeStart(HWND window, HMONITOR monitor, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+    void MoveSizeUpdate(HMONITOR monitor, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+    void MoveSizeEnd(HWND window, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+
+    void MoveWindowIntoZoneByIndex(HWND window, HMONITOR monitor, int index, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+    bool MoveWindowIntoZoneByDirection(HMONITOR monitor, HWND window, DWORD vkCode, bool cycle, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap);
+
+private:
+    class WindowMoveHandlerPrivate* pimpl;

```

Comment:
What are the benefits of using pimpl here?

Reply:
The only benefit is to hide private methods. If you think that it's overhead I can remove it.

Reply:
I don't believe in encapsulation, but it's fine for me either way!

Reply:
Does it have any performance impact? If not leave it as it is.

Reply:
It doesn't have performance impact, there are just private function calls. So I'll leave it as it is.